test/test_unit.py

```diff
@@ -1,43 +1,10 @@
 import pytest
 from pydantic import ValidationError
-from steamship import Block, SteamshipError, TaskState
-from steamship.plugin.inputs.raw_block_and_tag_plugin_input import RawBlockAndTagPluginInput
-from steamship.plugin.request import PluginRequest
+from steamship import SteamshipError
 
 from api import DallEPlugin, ImageSizeEnum
 
 
-def test_generator():
-    dalle = DallEPlugin()
-    request = PluginRequest(
-        data=RawBlockAndTagPluginInput(blocks=[Block(text="a cat riding a bicycle")])
-    )
-    response = dalle.run(request)
-
-    assert response.status.state is TaskState.succeeded
-    assert response.data is not None
-
-    assert len(response.data.blocks) == 1
-    assert response.data.blocks[0].url is not None
-
-
-def test_generator_overrides():
-    dalle = DallEPlugin()
-    request = PluginRequest(
-        data=RawBlockAndTagPluginInput(
-            blocks=[Block(text="a cat riding a bicycle")], options={"n": 2, "size": "512x512"}
-        )
-    )
-    response = dalle.run(request)
-
-    assert response.status.state is TaskState.succeeded
-    assert response.data is not None
-
-    assert len(response.data.blocks) == 2
-    for block in response.data.blocks:
-        assert block.url is not None
-
-
```

Comment:
question: Were these tests dropped because they don't fit the streaming generator model?  should we have a small harness in the SDK that allows easy unit testing of streaming generators?

Reply:
yeah, i didn't see a clear way to preserving these tests that added any real value (in a unit testy sort of way). counting blocks received just seemed like i'd be testing the engine, and waiting for streaming to complete seemed like the integration tests. happy to put something else in, if you have suggestions.

Reply:
In the gpt-4 tests, I faked the engine's allocation of blocks, and then waited on the streamed response, so that you could still test / debug locally before deploying.

https://github.com/steamship-plugins/gpt4/blob/5a41b6870af1f595923b0c21eccdb350b896b28a/test/test_unit.py#L254